rewards/src/lib.rs

```diff
@@ -315,6 +319,53 @@ impl<T: Config> Pallet<T> {
 		});
 	}
 
+	/// Splits share into two parts.
+	///
+	/// `move_share` - amount of share to remove and put into `other` share
+	/// `other` - new account who will own new share
+	///
+	/// Similar too claim and add 2 shares later, but does not requires pool
+	/// inflation and is more efficient.
+	pub fn transfer_share_and_rewards(
+		who: &T::AccountId,
+		pool: &T::PoolId,
+		move_share: T::Share,
+		other: &T::AccountId,
+	) -> DispatchResult {
+		SharesAndWithdrawnRewards::<T>::mutate(pool, other, |increased_share| {
+			let (increased_share, increased_rewards) = increased_share;
+			SharesAndWithdrawnRewards::<T>::mutate_exists(pool, who, |share| {
+				if let Some((share, rewards)) = share {
+					if move_share < *share {
+						for (reward_currency, balance) in rewards {
+							let move_balance = U256::from(balance.to_owned().saturated_into::<u128>())
+								* U256::from(move_share.to_owned().saturated_into::<u128>())
+								/ U256::from(share.to_owned().saturated_into::<u128>());
```

Comment:
need some comment to explain why this is safe

Reply:
explained some, so i feel back about that Share is not capped to some number so and saturated used in many places (thins will get broken if share is > u128, not only in this method so). i feel like Share needs more traits. they are not related mbr or u256, but converte into u128 said first.

Reply:
so this method seems as safe as others, until replace all sat_into_u128 with try_into or extend share traits. will close comment for now if that is ok